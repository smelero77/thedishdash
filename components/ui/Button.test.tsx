import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Button } from './Button';

describe('<Button />', () => {
  const handleClick = jest.fn();

  beforeEach(() => {
    handleClick.mockClear();
  });

  it('renderiza con props por defecto', () => {
    render(<Button onClick={handleClick}>Texto</Button>);
    const btn = screen.getByRole('button', { name: /texto/i }) as HTMLButtonElement;
    expect(btn).toBeInTheDocument();
  });

  it('aplica className extra', () => {
    render(
      <Button onClick={handleClick} className="mi-clase">
        Hola
      </Button>
    );
    const btn = screen.getByRole('button', { name: /hola/i });
    expect(btn).toHaveClass('mi-clase');
  });

  it('llama a onClick al hacer click', async () => {
    render(<Button onClick={handleClick}>Click me</Button>);
    await userEvent.click(screen.getByRole('button', { name: /click me/i }));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('no llama a onClick si estÃ¡ deshabilitado', async () => {
    render(
      <Button onClick={handleClick} disabled>
        Deshabilitado
      </Button>
    );
    const btn = screen.getByRole('button', { name: /deshabilitado/i });
    expect(btn).toBeDisabled();
    await userEvent.click(btn);
    expect(handleClick).not.toHaveBeenCalled();
  });

  it('coincide con el snapshot', () => {
    const { asFragment } = render(
      <Button onClick={handleClick} className="otra-clase">
        Snapshot
      </Button>
    );
    expect(asFragment()).toMatchSnapshot();
  });
}); 