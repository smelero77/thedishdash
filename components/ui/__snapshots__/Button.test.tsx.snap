// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> coincide con el snapshot 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2 otra-clase"
  >
    Snapshot
  </button>
</DocumentFragment>
`;
